spec/rubocop/cop/style/double_pipe_equals_spec.rb

```diff
@@ -0,0 +1,94 @@
+# frozen_string_literal: true
+
+describe RuboCop::Cop::Style::DoublePipeEquals do
+  let(:config) { RuboCop::Config.new }
+  subject(:cop) { described_class.new(config) }
+
+  context 'when using var = var ? var : something' do
+    it 'registers an offense' do
+      expect_offense(<<-RUBY.strip_indent)
+        foo = foo ? foo : 'default'
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^^ Use the double pipe equals operator `||=` instead.
+      RUBY
+
+      expect_offense(<<-RUBY.strip_indent)
+        foo = if foo
+        ^^^^^^^^^^^^ Use the double pipe equals operator `||=` instead.
+                foo
+              else
+                'default'
+              end
+      RUBY
+    end
+
+    it 'autocorrects to `var ||= something`' do
+      new_source = autocorrect_source('foo = foo ? foo : 3')
+      expect(new_source).to eq('foo ||= 3')
+
+      new_source2 = autocorrect_source(<<-RUBY.strip_indent)
+        foo = if foo
+                foo
+              else
+                3
+              end
+      RUBY
+      expect(new_source2).to eq('foo ||= 3')
+    end
+
+    it 'does not register an offense if any of the variables are different' do
+      expect_no_offenses('foo = bar ? foo : 3')
+      expect_no_offenses('foo = foo ? bar : 3')
+      expect_no_offenses(<<-RUBY.strip_indent)
+        foo = if foo
+                bar
+              else
+                3
+              end
+      RUBY
+      expect_no_offenses(<<-RUBY.strip_indent)
+        foo = if bar
+                foo
+              else
+                3
+              end
+      RUBY
+    end
+  end
+
+  context 'when using var = something unless var' do
+    it 'registers an offense' do
+      expect_offense(<<-RUBY.strip_indent)
+        foo = 'default' unless foo
+        ^^^^^^^^^^^^^^^^^^^^^^^^^^ Use the double pipe equals operator `||=` instead.
+      RUBY
+
+      expect_offense(<<-RUBY.strip_indent)
```

Comment:
```
Failures:

  1) RuboCop::Cop::Style::DoublePipeEquals when using var = something unless var registers an offense
     Failure/Error: expect(actual_annotations.to_s).to eq(expected_annotations.to_s)
     
       expected: "unless foo\n^^^^^^^^^^ Use the double pipe equals operator `||=` instead.\n  foo = 'default'\nend\n"
            got: "unless foo\n^^^^^^^^^^ Use the double pipe equals operator `||=` instead.\n^^^^^^^^^^^^^^^^^^^^^^^^^^ Use the double pipe equals operator `||=` instead.\n  foo = 'default'\nend\n"
     
       (compared using ==)
     
       Diff:
       @@ -1,5 +1,6 @@
        unless foo
        ^^^^^^^^^^ Use the double pipe equals operator `||=` instead.
       +^^^^^^^^^^^^^^^^^^^^^^^^^^ Use the double pipe equals operator `||=` instead.
          foo = 'default'
        end
```

Hmm... I guess it may be `expect_offense` bug. This helper method does not seem to support offense that is on multiple lines.  cc: @backus 
Could you use `inspect_source` instead of? Ref: https://github.com/bbatsov/rubocop/blob/27b9287c46a50436d8d0aa905855ad43b20ed3a5/spec/rubocop/cop/layout/access_modifier_indentation_spec.rb#L17-L27